docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+# Conditional code
+
+Webpack's static code analyzer is capable of removing dead code branch from the bundles it creates. The Dojo build tool leverages this capability to create branches of conditional code that can be removed at build time. Named conditional blocks are defined using Dojo framework's `has` module. Then those conditional blocks can be statically set to true or false through `.dojorc` and removed at build time.
+
+> main.ts
+
+```ts
+import has from '@dojo/framework/has';
+
+if (has('production')) {
+	console.log('Starting in production');
+} else {
+	console.log('Starting in dev mode');
+}
+
+export const mode = has('production') ? 'dist' : 'dev';
+```
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"features": {
+			"production": true
+		}
+	}
+}
+```
+
+The above `production` feature will be set `true` for **production builds** (`dist` mode). The Dojo build system uses the [static-build-loader](https://github.com/dojo/webpack-contrib/#static-build-loader) to rewrite `has()` calls that have been statically asserted by the system as `true` or `false`. This allows Webpack to conditionally identify code as unreachable and remove those dead code branches from the build.
+
+For example, the above code would be rewritten as:
+
+> static-build-loader output
+
+```js
+import has from '@dojo/framework/has';
+
+if (true) {
+	console.log('Starting in production');
+} else {
+	console.log('Starting in dev mode');
+}
+
+export const mode = true ? 'dist' : 'dev';
+```
+
+Webpack's dead branch removal would then remove the unreachable code.
+
+> Uglify output
+
+```js
+console.log('Starting in production');
+export const mode = 'dist';
+```
+
+Any features which are not statically asserted, are not re-written. This allows the code to determine at run-time if the feature is present.
+
+## Elided Imports
+
+The Dojo build process allows for imports to be conditionally removed through the use of _has pragmas_. These pragmas are simply strings that refer to a specific feature. If the feature is statically set at build time then the build system will evaluate the pragma and if `true` the next import is removed.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"features": {
+			"foo": true,
+			"bar": false
+		}
+	}
+}
+```
+
+> main.ts
+
+```ts
+"has('foo')";
+import 'a'; // removed
+'!has("bar")';
+import 'b'; // removed
+'!has("foo")';
+import 'c';
+'has("baz")';
+import 'd';
+```
+
+In the above example, imports `a` and `b` would be removed because their values evaluate to `true`. Import `c` will remain because it evaluates to `false`. Import `d` will remain because the `baz` feature has not been statically defined.
+
+## Known features
```

Comment:
I don't think this is correct anymore, we simply have `modern` - however I am not sure this is something that a user should really need to know.